docs/source/notes/libtorch_stable_abi.md

```diff
@@ -0,0 +1,34 @@
+# LibTorch Stable ABI
+
+This note will eventually contain more details on how to use the APIs in torch/csrc/stable. For the moment, it contains a table of internal representations:
+1. type in custom extension: type used within the end user custom library.
+2. StableIValue representation: a stable conversion of the type to liaison between the user model vs libtorch.so in an ABI-stable manner.
+3. type in libtorch: type used within libtorch.so (or any code binary locked with libtorch).
+4. Schema Type: type as described by the schema, which we hail as the source of truth for both ATen ops in native_functions.yaml and for user defined custom operators registered to the dispatcher via TORCH_LIBRARY or torch.library.
+
+|  type in custom extension    |   StableIValue representation   |   type in libtorch  |   Schema Type  |
+| -------- | ------- | ------- | ------- |
+| std::optional\<S> | \*reinterpret_cast\<(StableIValue\*)\*>, pointer to a StableIValue recursively defined | std::optional\<T> | Type? |
+| std::nullopt | \*reinterpret_cast\<nullptr_t\*> | IValue() | None |
+| RAIIATH | \*reinterpret_cast\<uint64_t\*> of AtenTensorHandle | at::Tensor |  Tensor |
+| int32_t | \*reinterpret_cast\<uint64_t\*> | at::ScalarType | ScalarType |
+| int32_t | \*reinterpret_cast\<uint64_t\*> | at::Layout | Layout |
+| int32_t | \*reinterpret_cast\<uint64_t\*> | at::MemoryFormat | MemoryFormat |
+| bool | \*reinterpret_cast\<uint64_t\*> | bool | bool |
+| int64_t | \*reinterpret_cast\<uint64_t\*> | int64_t | int |
```

Comment:
The way I am reading this is that you took an int64_t val, did a `reinterpret_cast<uint64_t*>(val)`, and then dereferenced it to get a uint64_t.
 
1. As written this is UB. So it probably means that we do a `reinterpret_cast<uint64_t*>(&val)`
2. Can this just be `reinterpret_cast<uint64_t>(val)`?

Reply:
Ah, in code it is the first, I had just failed to document correctly. Sorry! - will fix

Reply:
2. no, reinterpret_cast cannot cast with floats, so it'll fail for a subset of these